### ISSUE TITLE
Feature Request: Unused bounds lint

### ISSUE THREAD

**Issue by c410-f3r** [rust-lang/rust-clippy#5719]
Couldn't find an issue about it here or in the Rust repository.

Basically, print a warn for bounds that are not being used by a type:

```rust
// Default bound isn't being used
fn foo<T: Default>(t: T) {
  let _do_nothing = t;
}
```

Not sure but it probably won't be a trivial implementation :)

**Comment by flip1995** [comment]
This may have different difficulties for different bounds. `Default` may be moderately easy, since it only requires a function body visitor that checks if `Default::default` is called on `T`.

This may get harder the more general cases this lint should check. I.e. `impl`s, `struct`s, ...

**Comment by c410-f3r** [comment]
Closing due to inactiveness
